models/DataObject/Fieldcollection/Definition.php

```diff
@@ -165,7 +165,7 @@ public function save($saveDefinitionFile = true)
         $cd .= 'use Pimcore\Model\DataObject\PreGetValueHookInterface;';
         $cd .= "\n\n";
 
-        $implementsParts = ['\\Pimcore\\Model\\DataObject\\DirtyIndicatorInterface'];
+        $implementsParts = ['\\Pimcore\\Model\\Element\\DirtyIndicatorInterface'];
```

Comment:
We should move that to the abstract class as well, just apply the interface on `\Pimcore\Model\DataObject\Fieldcollection\Data\AbstractData` then this can go away. 

Same for the trait and the object bricks (missing at the moment), see 
https://github.com/pimcore/pimcore/blob/663b0887cf4623b5499987e2b15b9522f3342e11/models/DataObject/Fieldcollection/Definition.php#L175
and
https://github.com/pimcore/pimcore/blob/cc1059ad1d4458aa410f9d46ffe279336993a941/models/DataObject/Objectbrick/Definition.php#L231

Reply:
changed for Fieldcollection\Definition, Objectbrick\Definition & DataObject\Classdefinition